src/controllers/groupController.ts

```diff
@@ -0,0 +1,161 @@
+import { Request, Response } from 'express';
+import { StatusCodes } from 'http-status-codes';
+import { GroupService } from '../services/groupService';
+import { IGroup } from './../helpers/dto';
+const groupServiceInstance = new GroupService();
+
+export class GroupController {
+  async createGroup(request: Request, response: Response): Promise<Response> {
+    try {
+      const group = await groupServiceInstance.createGroup({
+        ...(request.body as IGroup)
+      });
+      return response.status(StatusCodes.OK).send(group);
+    } catch (err) {
+      if (err instanceof Error) {
+        return response
+          .status(StatusCodes.INTERNAL_SERVER_ERROR)
+          .json({ error: err.message });
+      }
+      const errorMessage = 'Failed to do something exceptional';
+      return response
+        .status(StatusCodes.INTERNAL_SERVER_ERROR)
+        .json({ error: errorMessage });
```

Comment:
No pior dos casos, caso o sugerido não funcione:

```suggestion
      return response
	        .status(StatusCodes.INTERNAL_SERVER_ERROR)
	        .json({ error: err instanceof Error ? err.message : 'Failed to do something exceptional'
```